apps/admin-x-activitypub/src/api/activitypub.ts

```diff
@@ -160,4 +160,52 @@ export class ActivityPubAPI {
         const url = new URL(`.ghost/activitypub/actions/unlike/${encodeURIComponent(id)}`, this.apiUrl);
         await this.fetchJSON(url, 'POST');
     }
+
+    get activitiesApiUrl() {
+        return new URL(`.ghost/activitypub/activities/${this.handle}`, this.apiUrl);
+    }
+
+    async getAllActivities(): Promise<Activity[]> {
+        const LIMIT = 50;
+
+        const fetchActivities = async (url: URL): Promise<Activity[]> => {
+            const json = await this.fetchJSON(url);
+
+            // If the response is null, return early
+            if (json === null) {
+                return [];
+            }
+
+            // If the response doesn't have an items array, return early
+            if (!('items' in json)) {
+                return [];
+            }
+
+            // If the response has an items property, but it's not an array
+            // use an empty array
+            const items = Array.isArray(json.items) ? json.items : [];
+
+            // If the response has a nextCursor property, fetch the next page
+            // recursively and concatenate the results
+            if ('nextCursor' in json && typeof json.nextCursor === 'string') {
```

Comment:
Will we do this dynamically eventually - or the plan is to stick with loading everything up front?

Reply:
Yeh it will be dynamic eventually, but for now we have no frontend pagination mechanism, so we load everything all at once (current behaviour)